client/signup/main.jsx

```diff
@@ -571,6 +584,18 @@ class Signup extends Component {
 		return window.location.origin + path;
 	};
 
+	getFilteredQueryArgs = () => {
+		const queryObject = this.props.initialContext?.query ?? {};
+		const flow = flows.getFlow( this.props.flowName, this.props.isLoggedIn );
+		const { siteId, siteSlug } = queryObject;
+
+		return {
+			siteId,
+			siteSlug,
+			...pick( queryObject, flow.providesDependenciesInQuery ),
```

Comment:
Since we are trying to avoid lodash as much as possible ( p4TIVU-9Bf-p2 ) maybe we can do this manually?

Reply:
Thanks for the reminder. I've removed it in https://github.com/Automattic/wp-calypso/pull/70371/commits/ffc63523704624a4b2de09d005ea7170557a822e. Along with this, it's an obvious opportunity to make `UNSAFE_componentWillMount()` a bit cleaner. I also used the chance to add `flags` as one of them. It is annonying that the feature flag is lost between navigation of signup steps :)